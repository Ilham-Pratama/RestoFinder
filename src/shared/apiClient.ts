import axios from 'axios';
import { YELP_API_TOKEN } from 'react-native-dotenv';

export const BASE_API_URL = 'https://api.yelp.com/v3';

export const apiClient = axios.create({
  baseURL: BASE_API_URL,
  timeout: 1000,
  headers: {
    Authorization: `Bearer ${YELP_API_TOKEN}`
  }
});
