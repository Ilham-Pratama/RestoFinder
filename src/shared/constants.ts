import { ICategoryItem } from 'components/CategoryItem/types';

export const MENU_CATEGORIES: ICategoryItem[] = [
  {
    label: 'Burger',
    value: 'burgers',
    image: require('assets/images/burger.png')
  },
  {
    label: 'Dessert',
    value: 'desserts',
    image: require('assets/images/cake.png')
  },
  {
    label: 'Pasta',
    value: 'pastashops',
    image: require('assets/images/pasta.png')
  },
  {
    label: 'Pizza',
    value: 'pizza',
    image: require('assets/images/pizza.png')
  },
  {
    label: 'Smoothies',
    value: 'juicebars',
    image: require('assets/images/smoothies.png')
  },
  {
    label: 'steak',
    value: 'steak',
    image: require('assets/images/steak.png')
  }
];
