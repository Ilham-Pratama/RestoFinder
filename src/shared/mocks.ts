export const BUSINESS_DUMMY = {
  id: 'WavvLdfdP6g8aZTtbBQHTw',
  alias: 'gary-danko-san-francisco',
  name: 'Gary Danko',
  image_url:
    'https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg',
  is_claimed: true,
  is_closed: false,
  url: 'https://www.yelp.com/biz/gary-danko-san-francisco?adjust_creative=wpr6gw4FnptTrk1CeT8POg&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_lookup&utm_source=wpr6gw4FnptTrk1CeT8POg',
  phone: '+14157492060',
  display_phone: '(415) 749-2060',
  review_count: 5296,
  categories: [
    {
      alias: 'newamerican',
      title: 'American (New)'
    },
    {
      alias: 'french',
      title: 'French'
    },
    {
      alias: 'wine_bars',
      title: 'Wine Bars'
    }
  ],
  rating: 4.5,
  location: {
    address1: '800 N Point St',
    address2: '',
    address3: '',
    city: 'San Francisco',
    zip_code: '94109',
    country: 'US',
    state: 'CA',
    display_address: ['800 N Point St', 'San Francisco, CA 94109'],
    cross_streets: ''
  },
  coordinates: {
    latitude: 37.80587,
    longitude: -122.42058
  },
  photos: [
    'https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg',
    'https://s3-media4.fl.yelpcdn.com/bphoto/FmXn6cYO1Mm03UNO5cbOqw/o.jpg',
    'https://s3-media4.fl.yelpcdn.com/bphoto/HZVDyYaghwPl2kVbvHuHjA/o.jpg'
  ],
  price: '$$$$',
  hours: [
    {
      open: [
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 0
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 1
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 2
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 3
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 4
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 5
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 6
        }
      ],
      hours_type: 'REGULAR',
      is_open_now: false
    }
  ],
  transactions: [],
  special_hours: [
    {
      date: '2019-02-07',
      is_closed: null,
      start: '1600',
      end: '2000',
      is_overnight: false
    }
  ]
};

export const REVIEW_DUMMY = {
  id: 'xAG4O7l-t1ubbwVAlPnDKg',
  rating: 5,
  user: {
    id: 'W8UK02IDdRS2GL_66fuq6w',
    profile_url:
      'https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w',
    image_url:
      'https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg',
    name: 'Ella A.'
  },
  text: 'Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...',
  time_created: '2016-08-29 00:41:13',
  url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
};

export const REVIEWS_DUMMY = {
  reviews: [
    {
      id: 'xAG4O7l-t1ubbwVAlPnDKg',
      rating: 5,
      user: {
        id: 'W8UK02IDdRS2GL_66fuq6w',
        profile_url:
          'https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w',
        image_url:
          'https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg',
        name: 'Ella A.'
      },
      text: 'Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...',
      time_created: '2016-08-29 00:41:13',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    },
    {
      id: '1JNmYjJXr9ZbsfZUAgkeXQ',
      rating: 4,
      user: {
        id: 'rk-MwIUejOj6LWFkBwZ98Q',
        profile_url:
          'https://www.yelp.com/user_details?userid=rk-MwIUejOj6LWFkBwZ98Q',
        image_url: null,
        name: 'Yanni L.'
      },
      text: 'The "restaurant" is inside a small deli so there is no sit down area. Just grab and go.\n\nInside, they sell individually packaged ingredients so that you can...',
      time_created: '2016-09-28 08:55:29',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=fj87uymFDJbq0Cy5hXTHIA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    },
    {
      id: 'SIoiwwVRH6R2s2ipFfs4Ww',
      rating: 4,
      user: {
        id: 'rpOyqD_893cqmDAtJLbdog',
        profile_url:
          'https://www.yelp.com/user_details?userid=rpOyqD_893cqmDAtJLbdog',
        image_url: null,
        name: 'Suavecito M.'
      },
      text: 'Dear Mission District,\n\nI miss you and your many delicious late night food establishments and vibrant atmosphere.  I miss the way you sound and smell on a...',
      time_created: '2016-08-10 07:56:44',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=m_tnQox9jqWeIrU87sN-IQ&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    }
  ],
  total: 3,
  possible_languages: ['en']
};
