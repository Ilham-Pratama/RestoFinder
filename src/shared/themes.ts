import { Theme } from '@react-navigation/native';
import { DarkTheme } from '@react-navigation/native';

export const AppLightTheme: Theme = {
  dark: false,
  colors: {
    primary: '#c3f229',
    background: '#F2F2F2',
    card: '#FFF',
    text: '#1C1C1E',
    border: '#E2FFA7',
    notification: '#C1FC4C'
  }
};

export const AppDarkTheme: Theme = {
  dark: true,
  colors: {
    ...DarkTheme.colors,
    primary: '#3D3D3D',
    card: '#1A1A1A',
    background: '#121212'
  }
};
