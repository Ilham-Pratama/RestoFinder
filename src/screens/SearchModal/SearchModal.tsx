import React from 'react';
import {
  FlatList,
  Pressable,
  SafeAreaView,
  StyleSheet,
  View
} from 'react-native';
import LocationIcon from 'assets/svgs/LocationIcon';
import SearchIcon from 'assets/svgs/SearchIcon';
import ShopIcon from 'assets/svgs/ShopIcon';
import CategoryItem from 'components/CategoryItem';
import MainTextInput from 'components/MainTextInput';
import ThemedButton from 'components/ThemedButton';
import { MENU_CATEGORIES } from 'shared/constants';
import sharedStyles from 'shared/styles';
import type { Category, NativeStackScreenProp } from 'types/globalTypes';

const SearchModal = ({
  route: { params },
  navigation: { setParams, navigate }
}: NativeStackScreenProp<'SearchModal'>) => {
  const { location, term, categories } = params;

  const onCategorySelect = (value: Category) => {
    if (categories.includes(value)) {
      const newCategories = categories.filter(v => v !== value);
      setParams({ categories: newCategories });
    } else setParams({ categories: [...categories, value] });
  };

  const setTerm = (text: string) => {
    setParams({ term: text });
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={sharedStyles.marginTop2}>
        <Pressable
          accessibilityRole="tab"
          onPress={() => navigate('SearchCityModal', params)}>
          <MainTextInput value={location} icon={LocationIcon} disabled />
        </Pressable>
      </View>
      <View style={sharedStyles.marginTop2}>
        <MainTextInput
          value={term}
          onChangeText={setTerm}
          icon={ShopIcon}
          placeholder="Search name, category, food, or anything"
        />
      </View>
      <View style={sharedStyles.marginTop1}>
        <FlatList
          data={MENU_CATEGORIES}
          keyExtractor={item => item.value}
          renderItem={({ item }) => (
            <CategoryItem
              {...item}
              onPress={onCategorySelect}
              isLastElement={
                item.value === MENU_CATEGORIES[MENU_CATEGORIES.length - 1].value
              }
              isActive={categories.includes(item.value)}
            />
          )}
          showsHorizontalScrollIndicator={false}
          horizontal
          accessibilityRole="list"
        />
      </View>
      <View style={styles.bottomContainer}>
        <ThemedButton
          title="Search"
          size="small"
          icon={SearchIcon}
          onPress={() => {
            navigate('Home', {
              term,
              categories,
              location
            });
          }}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  bottomContainer: {
    flex: 1,
    justifyContent: 'flex-end'
  }
});

export default SearchModal;
