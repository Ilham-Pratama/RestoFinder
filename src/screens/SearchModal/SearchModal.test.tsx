import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import SearchModal from '.';
import { MENU_CATEGORIES } from 'shared/constants';

describe('SearchModal', () => {
  const params = {
    location: 'Houston, Texas',
    term: 'Sample term',
    categories: [MENU_CATEGORIES[0].value]
  };
  it('renders SearchModal', () => {
    const setParams = jest.fn();
    const navigate = jest.fn();

    const route: any = { params };
    const navigation: any = { setParams, navigate };

    const component = render(
      <SearchModal route={route} navigation={navigation} />
    );

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getByText(params.location).props).toHaveProperty(
      'accessibilityRole',
      'text'
    );

    fireEvent.press(component.getByRole('tab'));

    expect(navigate).toBeCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith('SearchCityModal', params);

    const newTerm = 'New term';

    fireEvent.changeText(component.getByRole('search'), newTerm);

    expect(setParams).toHaveBeenCalledWith({ term: newTerm });
    expect(component.getByRole('list').props.data).toHaveLength(
      MENU_CATEGORIES.length
    );

    fireEvent.press(component.getAllByRole('button')[0]);

    expect(setParams).toHaveBeenCalledWith({ categories: [] });

    fireEvent.press(component.getAllByRole('button')[2]);

    expect(setParams).toHaveBeenCalledWith({
      categories: [...params.categories, MENU_CATEGORIES[2].value]
    });

    // Get and click the bottom button
    fireEvent.press(component.getAllByRole('button')[MENU_CATEGORIES.length]);

    expect(navigate).toBeCalledTimes(2);
    expect(navigate).toHaveBeenCalledWith('Home', params);
  });
});
