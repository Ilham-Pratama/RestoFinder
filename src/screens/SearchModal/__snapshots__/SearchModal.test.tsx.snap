// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchModal renders SearchModal 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityRole="tab"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 30,
            "elevation": 3,
            "flexDirection": "row",
            "marginHorizontal": 20,
            "paddingHorizontal": 12,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 5,
              "width": 5,
            },
            "shadowOpacity": 0.1,
          }
        }
      >
        <RNSVGSvgView
          accessibilityRole="image"
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          fill="#000"
          focusable={false}
          height={20}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          vbHeight={512}
          vbWidth={384}
          width={20}
        >
          <RNSVGGroup
            fill={4278190080}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M384 192C384 279.4 267 435 215.7 499.2C203.4 514.5 180.6 514.5 168.3 499.2C116.1 435 0 279.4 0 192C0 85.96 85.96 0 192 0C298 0 384 85.96 384 192H384z"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          accessibilityRole="text"
          style={
            Array [
              Object {
                "marginLeft": 15,
                "marginVertical": 15,
                "paddingRight": 10,
              },
              Object {
                "color": "black",
              },
            ]
          }
        >
          Houston, Texas
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 30,
          "elevation": 3,
          "flexDirection": "row",
          "marginHorizontal": 20,
          "paddingHorizontal": 12,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 0.1,
        }
      }
    >
      <RNSVGSvgView
        accessibilityRole="image"
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        fill="#000"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={512}
        vbWidth={640}
        width={20}
      >
        <RNSVGGroup
          fill={4278190080}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M0 155.2C0 147.9 2.153 140.8 6.188 134.7L81.75 21.37C90.65 8.021 105.6 0 121.7 0H518.3C534.4 0 549.3 8.021 558.2 21.37L633.8 134.7C637.8 140.8 640 147.9 640 155.2C640 175.5 623.5 192 603.2 192H36.84C16.5 192 .0003 175.5 .0003 155.2H0zM64 224H128V384H320V224H384V464C384 490.5 362.5 512 336 512H112C85.49 512 64 490.5 64 464V224zM512 224H576V480C576 497.7 561.7 512 544 512C526.3 512 512 497.7 512 480V224z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <TextInput
        accessibilityRole="search"
        icon={[Function]}
        onChangeText={[Function]}
        placeholder="Search name, category, food, or anything"
        placeholderTextColor="gray"
        style={
          Array [
            Object {
              "color": "black",
              "flex": 1,
            },
            Object {
              "marginLeft": 10,
            },
          ]
        }
        value="Sample term"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <RCTScrollView
      accessibilityRole="list"
      data={
        Array [
          Object {
            "image": Object {
              "process": [Function],
            },
            "label": "Burger",
            "value": "burgers",
          },
          Object {
            "image": Object {
              "process": [Function],
            },
            "label": "Dessert",
            "value": "desserts",
          },
          Object {
            "image": Object {
              "process": [Function],
            },
            "label": "Pasta",
            "value": "pastashops",
          },
          Object {
            "image": Object {
              "process": [Function],
            },
            "label": "Pizza",
            "value": "pizza",
          },
          Object {
            "image": Object {
              "process": [Function],
            },
            "label": "Smoothies",
            "value": "juicebars",
          },
          Object {
            "image": Object {
              "process": [Function],
            },
            "label": "steak",
            "value": "steak",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityRole="checkbox"
            style={
              Array [
                Object {
                  "borderRadius": 35,
                  "height": 100,
                  "marginLeft": 15,
                  "marginVertical": 10,
                  "overflow": "hidden",
                  "width": 70,
                },
                Object {
                  "elevation": 3,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 5,
                  },
                  "shadowOpacity": 0.1,
                },
                Object {
                  "backgroundColor": "rgb(0, 122, 255)",
                  "marginRight": 0,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 25,
                      "height": 45,
                      "justifyContent": "center",
                      "width": 45,
                    },
                    Object {
                      "backgroundColor": "rgb(255, 255, 255)",
                    },
                  ]
                }
              >
                <Image
                  accessibilityRole="image"
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <Text
                accessibilityRole="text"
                style={
                  Array [
                    Object {
                      "marginTop": 4,
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                    },
                  ]
                }
              >
                Burger
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityRole="checkbox"
            style={
              Array [
                Object {
                  "borderRadius": 35,
                  "height": 100,
                  "marginLeft": 15,
                  "marginVertical": 10,
                  "overflow": "hidden",
                  "width": 70,
                },
                Object {
                  "elevation": 3,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 5,
                  },
                  "shadowOpacity": 0.1,
                },
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "marginRight": 0,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 25,
                      "height": 45,
                      "justifyContent": "center",
                      "width": 45,
                    },
                    Object {
                      "backgroundColor": "rgb(255, 255, 255)",
                    },
                  ]
                }
              >
                <Image
                  accessibilityRole="image"
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <Text
                accessibilityRole="text"
                style={
                  Array [
                    Object {
                      "marginTop": 4,
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                    },
                  ]
                }
              >
                Dessert
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityRole="checkbox"
            style={
              Array [
                Object {
                  "borderRadius": 35,
                  "height": 100,
                  "marginLeft": 15,
                  "marginVertical": 10,
                  "overflow": "hidden",
                  "width": 70,
                },
                Object {
                  "elevation": 3,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 5,
                  },
                  "shadowOpacity": 0.1,
                },
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "marginRight": 0,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 25,
                      "height": 45,
                      "justifyContent": "center",
                      "width": 45,
                    },
                    Object {
                      "backgroundColor": "rgb(255, 255, 255)",
                    },
                  ]
                }
              >
                <Image
                  accessibilityRole="image"
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <Text
                accessibilityRole="text"
                style={
                  Array [
                    Object {
                      "marginTop": 4,
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                    },
                  ]
                }
              >
                Pasta
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityRole="checkbox"
            style={
              Array [
                Object {
                  "borderRadius": 35,
                  "height": 100,
                  "marginLeft": 15,
                  "marginVertical": 10,
                  "overflow": "hidden",
                  "width": 70,
                },
                Object {
                  "elevation": 3,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 5,
                  },
                  "shadowOpacity": 0.1,
                },
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "marginRight": 0,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 25,
                      "height": 45,
                      "justifyContent": "center",
                      "width": 45,
                    },
                    Object {
                      "backgroundColor": "rgb(255, 255, 255)",
                    },
                  ]
                }
              >
                <Image
                  accessibilityRole="image"
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <Text
                accessibilityRole="text"
                style={
                  Array [
                    Object {
                      "marginTop": 4,
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                    },
                  ]
                }
              >
                Pizza
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityRole="checkbox"
            style={
              Array [
                Object {
                  "borderRadius": 35,
                  "height": 100,
                  "marginLeft": 15,
                  "marginVertical": 10,
                  "overflow": "hidden",
                  "width": 70,
                },
                Object {
                  "elevation": 3,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 5,
                  },
                  "shadowOpacity": 0.1,
                },
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "marginRight": 0,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 25,
                      "height": 45,
                      "justifyContent": "center",
                      "width": 45,
                    },
                    Object {
                      "backgroundColor": "rgb(255, 255, 255)",
                    },
                  ]
                }
              >
                <Image
                  accessibilityRole="image"
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <Text
                accessibilityRole="text"
                style={
                  Array [
                    Object {
                      "marginTop": 4,
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                    },
                  ]
                }
              >
                Smoothies
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityRole="checkbox"
            style={
              Array [
                Object {
                  "borderRadius": 35,
                  "height": 100,
                  "marginLeft": 15,
                  "marginVertical": 10,
                  "overflow": "hidden",
                  "width": 70,
                },
                Object {
                  "elevation": 3,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 5,
                  },
                  "shadowOpacity": 0.1,
                },
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "marginRight": 15,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "paddingTop": 10,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 25,
                      "height": 45,
                      "justifyContent": "center",
                      "width": 45,
                    },
                    Object {
                      "backgroundColor": "rgb(255, 255, 255)",
                    },
                  ]
                }
              >
                <Image
                  accessibilityRole="image"
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </View>
              <Text
                accessibilityRole="text"
                style={
                  Array [
                    Object {
                      "marginTop": 4,
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                    },
                  ]
                }
              >
                steak
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "elevation": 3,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 5,
              "width": 5,
            },
            "shadowOpacity": 0.1,
          },
          Object {
            "backgroundColor": "rgb(0, 122, 255)",
            "borderRadius": 20,
            "margin": 10,
            "paddingVertical": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "color": "rgb(28, 28, 30)",
              "marginRight": 10,
            },
          ]
        }
      >
        SEARCH
      </Text>
      <RNSVGSvgView
        accessibilityRole="image"
        align="xMidYMid"
        bbHeight={15}
        bbWidth={15}
        fill="rgb(28, 28, 30)"
        focusable={false}
        height={15}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 15,
              "width": 15,
            },
          ]
        }
        vbHeight={512}
        vbWidth={512}
        width={15}
      >
        <RNSVGGroup
          fill={4280032286}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</RCTSafeAreaView>
`;
